병헌/BOJ_01647_도시분할계획/BOJ_01647_kruskal.java

```diff
@@ -0,0 +1,100 @@
+package boj.gold5.BOJ_01647_도시분할계획;
+
+/**
+
+- @author 이병헌
+- @since 8/5/24
+- @see https://www.acmicpc.net/problem/01647
+- @git https://github.com/Hunnibs
+- @youtube
+- @performance
+- @category # kruskal # union-find
+- @note
+
+ */
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.StringTokenizer;
+
+public class BOJ_01647_kruskal {
+    private static int[] parent;
+
+    public static void main(String[] args) throws Exception{
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st;
+
+        st = new StringTokenizer(br.readLine());
+        int N = Integer.parseInt(st.nextToken());
+        int M = Integer.parseInt(st.nextToken());
+
+        List<Info> graph = new ArrayList<>();
+
+        for (int i = 0; i < M; i++){
+            st = new StringTokenizer(br.readLine());
+            int a = Integer.parseInt(st.nextToken());
+            int b = Integer.parseInt(st.nextToken());
+            int c = Integer.parseInt(st.nextToken());
+
+            graph.add(new Info(a, b, c));
+        }
+
+        Collections.sort(graph);
+
+        parent = new int[N+1];
+        for (int i = 0; i <= N; i++) {
+            parent[i] = i;
+        }
+
+        long answer = 0;
+        int max = 0;
+        for(Info info : graph){
+            if (find(info.from) != find(info.to)){
+                answer += info.weight;
+                union(info.from, info.to);
+                max = Math.max(max, info.weight);
+            }
+        }
```

Comment:
결국 MST를 구성하는 문제니까 Union의 회수가 N-1번 됐을 때 반복문 탈출하는 것이 조금 더 효율적일 것 같습니다.
```suggestion
       int count = 0;
        for(Info info : graph){
            if (find(info.from) != find(info.to)){
                answer += info.weight;
                union(info.from, info.to);
                max = Math.max(max, info.weight);
                if(++count == n-1) break;
                
            }
        }
```

Reply:
100MS가 줄었네요. 사소한 변경이지만 성능을 끌어올릴 수 있군효